# Import libraries

In [1]:
import pandas as pd
import numpy as np

In [2]:
labeled = pd.read_csv('data/labeled_data.csv')

moldset_labeled_cn7 = pd.read_csv('data/moldset_labeled_cn7.csv')
moldset_labeled_rg3 = pd.read_csv('data/moldset_labeled_rg3.csv')
moldset_labeled = pd.read_csv('data/moldset_labeled.csv')

moldset_unlabeled_cn7 = pd.read_csv('data/moldset_unlabeled_cn7.csv')
moldset_unlabeled_rg3 = pd.read_csv('data/moldset_unlabeled_rg3.csv')

supervised_cn7 = pd.read_csv('data/supervised_label_cn7.csv')

unlabeled = pd.read_csv('data/unlabeled_data.csv')

In [3]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)
pd.set_option('display.max_rows', None)

# Load data

In [4]:
# shape, info, head를 출력하는 함수

def inspect_df(df, name):
    print("=" * 80)
    print(f"📌 {name}")
    print("=" * 80)

    # Shape
    print(f"\n[Shape]")
    print(df.shape)

    # Info
    print(f"\n[Info]")
    df.info()

    # Head
    display(df.head())

    print("\n")

### `labeled_data.csv`

In [5]:
inspect_df(labeled, "Labeled Data")

📌 Labeled Data

[Shape]
(7996, 45)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 7996 entries, 0 to 7995
Data columns (total 45 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   _id                       7996 non-null   str    
 1   TimeStamp                 7996 non-null   str    
 2   PART_FACT_PLAN_DATE       7996 non-null   str    
 3   PART_FACT_SERIAL          7996 non-null   int64  
 4   PART_NAME                 7996 non-null   str    
 5   EQUIP_CD                  7996 non-null   str    
 6   EQUIP_NAME                7996 non-null   str    
 7   PassOrFail                7996 non-null   str    
 8   Reason                    71 non-null     str    
 9   Injection_Time            7996 non-null   float64
 10  Filling_Time              7996 non-null   float64
 11  Plasticizing_Time         7996 non-null   float64
 12  Cycle_Time                7996 non-null   float64
 13  Clamp_Close_Time          7996 

,_id,TimeStamp,PART_FACT_PLAN_DATE,PART_FACT_SERIAL,PART_NAME,EQUIP_CD,EQUIP_NAME,PassOrFail,Reason,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Switch_Over_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Barrel_Temperature_7,Hopper_Temperature,Mold_Temperature_1,Mold_Temperature_2,Mold_Temperature_3,Mold_Temperature_4,Mold_Temperature_5,Mold_Temperature_6,Mold_Temperature_7,Mold_Temperature_8,Mold_Temperature_9,Mold_Temperature_10,Mold_Temperature_11,Mold_Temperature_12
0,5f8928bb9c0189cc666ef19b,2020-10-16 04:57:47,2020-10-16 오전 12:00:00,24,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,Y,NaN,9.59,4.47,16.92,59.520000,7.13,653.409973,0.0,68.849998,647.98999,55.400002,30.700001,292.5,141.800003,136.800003,37.400002,59.299999,276.500000,274.700012,274.799988,269.200012,255.000000,229.699997,0.0,66.300003,0.0,0.0,24.799999,27.5,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
1,5f8928de9c0189cc666ef20b,2020-10-16 04:58:48,2020-10-16 오전 12:00:00,24,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,Y,NaN,9.60,4.48,16.91,59.580002,7.13,653.409973,0.0,68.839996,647.98999,55.299999,30.799999,292.5,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,0.0,67.199997,0.0,0.0,24.799999,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
2,5f8928df9c0189cc666ef213,2020-10-16 04:58:48,2020-10-16 오전 12:00:00,23,CN7 W/S SIDE MLD'G LH,S14,650톤-우진2호기,Y,NaN,9.60,4.48,16.91,59.580002,7.13,653.409973,0.0,68.839996,647.98999,55.299999,30.799999,292.5,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,0.0,67.199997,0.0,0.0,24.799999,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
3,5f8928f39c0189cc666ef25e,2020-10-16 04:59:48,2020-10-16 오전 12:00:00,23,CN7 W/S SIDE MLD'G LH,S14,650톤-우진2호기,Y,NaN,9.59,4.48,16.91,59.560001,7.13,653.419983,0.0,68.839996,647.98999,55.299999,31.000000,292.5,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,0.0,66.900002,0.0,0.0,25.000000,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
4,5f8928f59c0189cc666ef265,2020-10-16 04:59:48,2020-10-16 오전 12:00:00,24,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,Y,NaN,9.59,4.48,16.91,59.560001,7.13,653.419983,0.0,68.839996,647.98999,55.299999,31.000000,292.5,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,0.0,66.900002,0.0,0.0,25.000000,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0


### `moldset_labeled_cn7.csv`, `moldset_labeled_rg3.csv`, `moldset_labeled.csv`

In [6]:
inspect_df(moldset_labeled_cn7, "Moldset Labeled CN7 Data")

📌 Moldset Labeled CN7 Data

[Shape]
(1211, 26)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 1211 entries, 0 to 1210
Data columns (total 26 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                1211 non-null   int64  
 1   PassOrFail                1211 non-null   int64  
 2   Injection_Time            1211 non-null   float64
 3   Filling_Time              1211 non-null   float64
 4   Plasticizing_Time         1211 non-null   float64
 5   Cycle_Time                1211 non-null   float64
 6   Clamp_Close_Time          1211 non-null   float64
 7   Cushion_Position          1211 non-null   float64
 8   Plasticizing_Position     1211 non-null   float64
 9   Clamp_Open_Position       1211 non-null   float64
 10  Max_Injection_Speed       1211 non-null   float64
 11  Max_Screw_RPM             1211 non-null   float64
 12  Average_Screw_RPM         1211 non-null   float64
 13  Max_Injection_Press

,Unnamed: 0,PassOrFail,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,0,0,1.679123,1.491397,-0.478017,0.361047,0.97858,-0.711164,1.532018,0.0,-1.152398,-0.795730,-0.132022,0.63521,1.256518,-0.916676,-0.533773,0.729073,-1.598068,-0.672350,-1.536620,0.008133,-1.352970,-1.780463,1.023094,1.287574
1,1,0,2.080358,1.894372,-0.527100,1.886261,0.97858,-0.711164,1.471608,0.0,-1.347095,0.049553,-0.132022,0.63521,1.573231,-0.740752,-0.533773,0.056153,0.622557,0.914107,0.504051,-0.915098,-2.263700,-0.874018,1.023094,1.337339
2,2,0,2.080358,1.894372,-0.527100,1.886261,0.97858,-0.711164,1.471608,0.0,-1.347095,0.049553,-0.132022,0.63521,1.573231,-0.740752,-0.533773,0.056153,0.622557,0.914107,0.504051,-0.915098,-2.263700,-0.874018,1.023094,1.337339
3,3,0,1.679123,1.894372,-0.527100,1.377857,0.97858,1.202502,1.471608,0.0,-1.347095,1.740153,-0.132022,0.63521,1.256518,-0.740752,-0.533773,0.729073,-0.765355,1.231418,0.886706,-0.299626,0.013158,-1.176164,1.166162,1.337339
4,4,0,1.679123,1.894372,-0.527100,1.377857,0.97858,1.202502,1.471608,0.0,-1.347095,1.740153,-0.132022,0.63521,1.256518,-0.740752,-0.533773,0.729073,-0.765355,1.231418,0.886706,-0.299626,0.013158,-1.176164,1.166162,1.337339


In [7]:
inspect_df(moldset_labeled_rg3, "Moldset Labeled RG3 Data")

📌 Moldset Labeled RG3 Data

[Shape]
(1182, 26)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 1182 entries, 0 to 1181
Data columns (total 26 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                1182 non-null   int64  
 1   PassOrFail                1182 non-null   int64  
 2   Injection_Time            1182 non-null   float64
 3   Filling_Time              1182 non-null   float64
 4   Plasticizing_Time         1182 non-null   float64
 5   Cycle_Time                1182 non-null   float64
 6   Clamp_Close_Time          1182 non-null   float64
 7   Cushion_Position          1182 non-null   float64
 8   Plasticizing_Position     1182 non-null   float64
 9   Clamp_Open_Position       1182 non-null   float64
 10  Max_Injection_Speed       1182 non-null   float64
 11  Max_Screw_RPM             1182 non-null   float64
 12  Average_Screw_RPM         1182 non-null   float64
 13  Max_Injection_Press

,Unnamed: 0,PassOrFail,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,1211,0,-0.029099,-1.192531,1.977737,0.249511,1.033078,2.867608,-1.755816,0.0,2.187583,0.677109,-0.421798,-2.077961,-1.184902,-1.330250,-1.306878,0.096629,-0.496175,-0.002081,0.047749,0.282727,-0.354184,0.183232,-1.393485,-1.576042
1,1212,0,-0.029099,-1.192531,1.977737,0.249511,1.033078,2.867608,-1.755816,0.0,2.187583,0.677109,-0.421798,-2.077961,-1.184902,-1.330250,-1.306878,0.096629,-0.496175,-0.002081,0.047749,0.282727,-0.354184,0.183232,-1.393485,-1.576042
2,1213,0,-0.029099,-1.192531,1.528778,0.249511,1.033078,2.075089,-0.192928,0.0,2.187583,0.677109,-0.421798,-1.421642,-0.974786,-1.019133,-0.841877,0.562844,1.442567,-0.002081,0.340262,1.850096,-0.851297,0.183232,-1.535968,-1.576042
3,1214,0,-0.029099,-1.192531,1.528778,0.249511,1.033078,2.075089,-0.192928,0.0,2.187583,0.677109,-0.421798,-1.421642,-0.974786,-1.019133,-0.841877,0.562844,1.442567,-0.002081,0.340262,1.850096,-0.851297,0.183232,-1.535968,-1.576042
4,1215,0,-0.029099,-1.192531,0.630880,-0.298708,1.033078,1.277707,-1.755816,0.0,1.402554,0.677109,2.655764,-0.765222,-0.554587,-0.708027,-1.539374,1.262095,-0.253813,-0.002081,0.925242,0.282727,-1.348485,0.284910,-1.535968,-1.576042


In [8]:
inspect_df(moldset_labeled, "Moldsest Labeled Data")

📌 Moldsest Labeled Data

[Shape]
(2607, 47)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 2607 entries, 0 to 2606
Data columns (total 47 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                2607 non-null   int64  
 1   _id                       2607 non-null   str    
 2   TimeStamp                 2607 non-null   str    
 3   PART_FACT_PLAN_DATE       2607 non-null   str    
 4   PART_FACT_SERIAL          2607 non-null   int64  
 5   PART_NO                   2607 non-null   str    
 6   PART_NAME                 2607 non-null   str    
 7   EQUIP_CD                  2607 non-null   str    
 8   EQUIP_NAME                2607 non-null   str    
 9   PassOrFail                2607 non-null   int64  
 10  Reason                    52 non-null     str    
 11  Injection_Time            2607 non-null   float64
 12  Filling_Time              2607 non-null   float64
 13  Plasticizing_Time     

,Unnamed: 0,_id,TimeStamp,PART_FACT_PLAN_DATE,PART_FACT_SERIAL,PART_NO,PART_NAME,EQUIP_CD,EQUIP_NAME,PassOrFail,Reason,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Switch_Over_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Barrel_Temperature_7,Hopper_Temperature,Mold_Temperature_1,Mold_Temperature_2,Mold_Temperature_3,Mold_Temperature_4,Mold_Temperature_5,Mold_Temperature_6,Mold_Temperature_7,Mold_Temperature_8,Mold_Temperature_9,Mold_Temperature_10,Mold_Temperature_11,Mold_Temperature_12
0,0,5f8928bb9c0189cc666ef19b,2020-10-16 04:57:47,2020-10-16 오전 12:00:00,24,86141AA000,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,0,NaN,9.59,4.47,16.92,59.520000,7.13,653.409973,0.0,68.849998,647.98999,55.400002,30.700001,292.5,141.800003,136.800003,37.400002,59.299999,276.500000,274.700012,274.799988,269.200012,255.000000,229.699997,0.0,66.300003,0.0,0.0,24.799999,27.5,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
1,1,5f8928de9c0189cc666ef20b,2020-10-16 04:58:48,2020-10-16 오전 12:00:00,24,86141AA000,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,0,NaN,9.60,4.48,16.91,59.580002,7.13,653.409973,0.0,68.839996,647.98999,55.299999,30.799999,292.5,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,0.0,67.199997,0.0,0.0,24.799999,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
2,2,5f8928df9c0189cc666ef213,2020-10-16 04:58:48,2020-10-16 오전 12:00:00,23,86131AA000,CN7 W/S SIDE MLD'G LH,S14,650톤-우진2호기,0,NaN,9.60,4.48,16.91,59.580002,7.13,653.409973,0.0,68.839996,647.98999,55.299999,30.799999,292.5,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,0.0,67.199997,0.0,0.0,24.799999,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
3,3,5f8928f39c0189cc666ef25e,2020-10-16 04:59:48,2020-10-16 오전 12:00:00,23,86131AA000,CN7 W/S SIDE MLD'G LH,S14,650톤-우진2호기,0,NaN,9.59,4.48,16.91,59.560001,7.13,653.419983,0.0,68.839996,647.98999,55.299999,31.000000,292.5,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,0.0,66.900002,0.0,0.0,25.000000,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
4,4,5f8928f59c0189cc666ef265,2020-10-16 04:59:48,2020-10-16 오전 12:00:00,24,86141AA000,CN7 W/S SIDE MLD'G RH,S14,650톤-우진2호기,0,NaN,9.59,4.48,16.91,59.560001,7.13,653.419983,0.0,68.839996,647.98999,55.299999,31.000000,292.5,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,0.0,66.900002,0.0,0.0,25.000000,27.6,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0


### `moldset_unlabeled_cn7.csv`, `moldset_unlabeled_rg3.csv`

In [9]:
inspect_df(moldset_unlabeled_cn7, "Moldset Unlabeled CN7 Data")

📌 Moldset Unlabeled CN7 Data

[Shape]
(35239, 25)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 35239 entries, 0 to 35238
Data columns (total 25 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                35239 non-null  int64  
 1   Injection_Time            35239 non-null  float64
 2   Filling_Time              35239 non-null  float64
 3   Plasticizing_Time         35239 non-null  float64
 4   Cycle_Time                35239 non-null  float64
 5   Clamp_Close_Time          35239 non-null  float64
 6   Cushion_Position          35239 non-null  float64
 7   Plasticizing_Position     35239 non-null  float64
 8   Clamp_Open_Position       35239 non-null  float64
 9   Max_Injection_Speed       35239 non-null  float64
 10  Max_Screw_RPM             35239 non-null  float64
 11  Average_Screw_RPM         35239 non-null  float64
 12  Max_Injection_Pressure    35239 non-null  float64
 13  Max_Switch_Ove

,Unnamed: 0,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,114610,1.346829,-2.768862,0.961909,1.843294,1.029767,1.031111,0.849819,0.861854,2.489188,1.025136,1.031548,0.494886,0.345152,0.183351,0.221579,1.015093,0.919876,0.846380,0.637023,2.099671,1.025241,-0.551566,0.756435,0.600810
1,114688,1.360046,-2.720306,0.963280,1.847956,1.029767,1.031048,0.863559,0.861854,2.446877,1.025136,1.031548,0.497896,0.348230,0.201286,0.228419,1.020221,0.988615,0.874453,0.703109,2.099671,1.074495,-0.503460,0.818567,0.700620
2,114919,1.355641,-2.720306,0.966020,1.843294,1.029767,1.031080,0.864246,0.861854,2.446877,1.025136,1.031548,0.497896,0.351308,0.171395,0.221579,1.014360,0.862594,0.874453,0.637023,2.293882,0.975987,-0.599672,0.694303,0.574194
3,115304,1.360046,-2.720306,0.960539,1.843294,1.029767,1.031143,0.864246,0.861854,2.455339,1.025136,1.031548,0.497896,0.332839,0.183351,0.221579,1.017290,0.919876,0.944634,0.785717,2.099671,1.058077,-0.599672,0.531206,0.394536
4,115582,1.360046,-2.720306,0.967391,1.847956,1.029767,1.031048,0.864933,0.861854,2.446877,1.018644,1.031548,0.497896,0.326683,0.165417,0.221579,1.018756,0.851137,0.916562,0.670066,2.177355,0.992405,-0.599672,0.585571,0.434460


In [10]:
inspect_df(moldset_unlabeled_rg3, "Moldset Unlabeled RG3 Data")

📌 Moldset Unlabeled RG3 Data

[Shape]
(35941, 25)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 35941 entries, 0 to 35940
Data columns (total 25 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                35941 non-null  int64  
 1   Injection_Time            35941 non-null  float64
 2   Filling_Time              35941 non-null  float64
 3   Plasticizing_Time         35941 non-null  float64
 4   Cycle_Time                35941 non-null  float64
 5   Clamp_Close_Time          35941 non-null  float64
 6   Cushion_Position          35941 non-null  float64
 7   Plasticizing_Position     35941 non-null  float64
 8   Clamp_Open_Position       35941 non-null  float64
 9   Max_Injection_Speed       35941 non-null  float64
 10  Max_Screw_RPM             35941 non-null  float64
 11  Average_Screw_RPM         35941 non-null  float64
 12  Max_Injection_Pressure    35941 non-null  float64
 13  Max_Switch_Ove

,Unnamed: 0,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,119515,0.90919,1.662808,-1.131352,-1.174407,-1.262902,-1.314850,-1.026514,1.362091,-1.203070,-1.179145,-1.297009,-1.233889,-0.546310,-1.165335,-1.154268,-1.310547,-0.666957,-0.395393,0.088306,0.041839,0.945920,0.087427,-1.276966,-1.225083
1,119547,0.90919,1.662808,-1.131352,-1.174407,-1.262902,-1.314850,-1.026514,1.362091,-1.200666,-1.179145,-1.297009,-1.233889,-0.550067,-1.161183,-1.154268,-1.310547,-0.674680,-0.403577,0.078025,0.029691,0.925301,0.107476,-1.276966,-1.225083
2,119618,0.90919,1.662808,-1.147464,-1.174407,-1.262902,-1.315176,-1.026514,1.362091,-1.200666,-1.179145,-1.297009,-1.233889,-0.546310,-1.169488,-1.154268,-1.310547,-0.636064,-0.395393,0.078025,0.066135,0.904682,0.107476,-1.276966,-1.225083
3,119781,0.90919,1.662808,-1.131352,-1.174407,-1.262902,-1.314850,-1.026514,1.362091,-1.193454,-1.179145,-1.297009,-1.231013,-0.548189,-1.165335,-1.154268,-1.310547,-0.674680,-0.395393,0.067743,0.029691,0.863445,0.127524,-1.276966,-1.225083
4,120044,0.90919,1.662808,-1.131352,-1.174407,-1.262902,-1.314850,-1.026514,1.362091,-1.159797,-1.179145,-1.297009,-1.233889,-0.553825,-1.161183,-1.154268,-1.310547,-0.674680,-0.395393,0.088306,0.041839,0.945920,0.147573,-1.276966,-1.225083


### `supervised_label_cn7.csv`

In [11]:
inspect_df(supervised_cn7, "Supervised Label CN7 Data")

📌 Supervised Label CN7 Data

[Shape]
(6736, 26)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 6736 entries, 0 to 6735
Data columns (total 26 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                6736 non-null   int64  
 1   PassOrFail                6736 non-null   int64  
 2   Injection_Time            6736 non-null   float64
 3   Filling_Time              6736 non-null   float64
 4   Plasticizing_Time         6736 non-null   float64
 5   Cycle_Time                6736 non-null   float64
 6   Clamp_Close_Time          6736 non-null   float64
 7   Cushion_Position          6736 non-null   float64
 8   Plasticizing_Position     6736 non-null   float64
 9   Clamp_Open_Position       6736 non-null   float64
 10  Max_Injection_Speed       6736 non-null   float64
 11  Max_Screw_RPM             6736 non-null   float64
 12  Average_Screw_RPM         6736 non-null   float64
 13  Max_Injection_Pres

,Unnamed: 0,PassOrFail,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Hopper_Temperature,Mold_Temperature_3,Mold_Temperature_4
0,0,0,9.60,4.48,16.91,59.580002,7.13,653.409973,68.839996,647.98999,55.299999,30.799999,292.500000,141.800003,136.899994,37.500000,59.299999,276.200012,275.500000,275.299988,270.799988,254.699997,229.500000,67.199997,24.799999,27.6
1,1,0,9.59,4.48,16.91,59.560001,7.13,653.419983,68.839996,647.98999,55.299999,31.000000,292.500000,141.800003,136.800003,37.500000,59.299999,276.500000,275.000000,275.399994,271.100006,254.899994,230.000000,66.900002,25.000000,27.6
2,2,0,9.58,4.46,16.90,59.580002,7.13,653.409973,68.839996,647.98999,55.599998,30.900000,292.500000,141.699997,136.399994,37.700001,59.400002,276.299988,275.299988,275.200012,271.399994,255.000000,230.000000,67.500000,25.000000,27.6
3,3,0,9.58,4.46,16.92,59.560001,7.13,653.409973,68.849998,647.98999,55.500000,30.600000,292.399994,141.800003,136.699997,37.400002,59.299999,275.799988,275.399994,275.000000,271.299988,255.000000,230.000000,67.000000,25.000000,27.6
4,4,0,9.57,4.45,16.91,59.520000,7.14,653.409973,68.830002,647.98999,55.700001,30.799999,292.500000,141.600006,136.399994,37.099998,59.099998,275.700012,274.899994,274.799988,270.799988,255.399994,230.100006,66.699997,24.799999,27.5


### `unlabeled_data.csv`

In [12]:
inspect_df(unlabeled, "Unlabeled Data")

📌 Unlabeled Data

[Shape]
(795315, 46)

[Info]
<class 'pandas.DataFrame'>
RangeIndex: 795315 entries, 0 to 795314
Data columns (total 46 columns):
 #   Column                    Non-Null Count   Dtype  
---  ------                    --------------   -----  
 0   Unnamed: 0                795315 non-null  int64  
 1   _id                       795315 non-null  str    
 2   TimeStamp                 795315 non-null  str    
 3   PART_FACT_PLAN_DATE       795315 non-null  str    
 4   PART_FACT_SERIAL          795315 non-null  int64  
 5   PART_NO                   795315 non-null  str    
 6   PART_NAME                 795315 non-null  str    
 7   EQUIP_CD                  795315 non-null  str    
 8   EQUIP_NAME                795315 non-null  str    
 9   ERR_FACT_QTY              795315 non-null  int64  
 10  Injection_Time            795315 non-null  float64
 11  Filling_Time              795315 non-null  float64
 12  Plasticizing_Time         795315 non-null  float64
 13  Cycle_Ti

,Unnamed: 0,_id,TimeStamp,PART_FACT_PLAN_DATE,PART_FACT_SERIAL,PART_NO,PART_NAME,EQUIP_CD,EQUIP_NAME,ERR_FACT_QTY,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Switch_Over_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Barrel_Temperature_7,Hopper_Temperature,Mold_Temperature_1,Mold_Temperature_2,Mold_Temperature_3,Mold_Temperature_4,Mold_Temperature_5,Mold_Temperature_6,Mold_Temperature_7,Mold_Temperature_8,Mold_Temperature_9,Mold_Temperature_10,Mold_Temperature_11,Mold_Temperature_12
0,0,5f9136166e37704af878240e,2020-01-02 00:59:22,2020-01-02 오전 12:00:00,2,87285Q5000,"SP2 CVR ROOF RACK CTR, RH",S01,1800TON-우진,11,3.18,2.66,13.14,0.00,12.30,9.9,10.0,59.9,0.0,35.6,0.0,0.0,103.2,102.8,0.0,0.0,0.0,261.8,270.4,271.1,273.5,260.3,18.8,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0
1,1,5f9136166e37704af878240f,2020-01-02 01:01:01,2020-01-02 오전 12:00:00,1,87275Q5000,"SP2 CVR ROOF RACK CTR, LH",S01,1800TON-우진,11,16.29,2.78,13.72,0.00,12.50,9.0,10.0,59.9,883.3,24.6,28.1,26.6,128.1,127.4,5.5,48.3,0.0,258.4,270.7,271.0,274.8,262.3,18.1,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0
2,2,5f9136166e37704af8782410,2020-01-02 01:01:01,2020-01-02 오전 12:00:00,2,87285Q5000,"SP2 CVR ROOF RACK CTR, RH",S01,1800TON-우진,11,16.29,2.78,13.72,0.00,12.50,9.0,10.0,59.9,883.3,24.6,28.1,26.6,128.1,127.4,5.5,48.3,0.0,258.4,270.7,271.0,274.8,262.3,18.1,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0
3,3,5f9136166e37704af8782411,2020-01-02 01:02:24,2020-01-02 오전 12:00:00,1,87275Q5000,"SP2 CVR ROOF RACK CTR, LH",S01,1800TON-우진,11,16.28,2.78,12.74,81.38,12.77,8.7,10.0,59.9,883.4,23.9,28.1,25.9,127.9,127.2,5.3,48.8,0.0,258.6,270.3,270.5,274.4,262.7,19.4,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0
4,4,5f9136166e37704af8782412,2020-01-02 01:02:24,2020-01-02 오전 12:00:00,2,87285Q5000,"SP2 CVR ROOF RACK CTR, RH",S01,1800TON-우진,11,16.28,2.78,12.74,81.38,12.77,8.7,10.0,59.9,883.4,23.9,28.1,25.9,127.9,127.2,5.3,48.8,0.0,258.6,270.3,270.5,274.4,262.7,19.4,0.0,0,0,0.0,0.0,0,0,0,0,0,0,0,0


# 데이터 확인

### `unnamed: 0` 컬럼 제거
labeled, labeled_cn7, labeled_rg3, moldset_labeled, unlabeled_cn7, unlabeled_rg3, supervised_cn7, unlabeled

In [13]:
print(moldset_labeled_cn7.shape, moldset_labeled_rg3.shape, moldset_labeled.shape, moldset_unlabeled_cn7.shape, moldset_unlabeled_rg3.shape, supervised_cn7.shape, unlabeled.shape)

moldset_labeled_cn7.drop(columns=['Unnamed: 0'], inplace=True)
moldset_labeled_rg3.drop(columns=['Unnamed: 0'], inplace=True)
moldset_labeled.drop(columns=['Unnamed: 0'], inplace=True)
moldset_unlabeled_cn7.drop(columns=['Unnamed: 0'], inplace=True)
moldset_unlabeled_rg3.drop(columns=['Unnamed: 0'], inplace=True)
supervised_cn7.drop(columns=['Unnamed: 0'], inplace=True)
unlabeled.drop(columns=['Unnamed: 0'], inplace=True)

print(moldset_labeled_cn7.shape, moldset_labeled_rg3.shape, moldset_labeled.shape, moldset_unlabeled_cn7.shape, moldset_unlabeled_rg3.shape, supervised_cn7.shape, unlabeled.shape)

(1211, 26) (1182, 26) (2607, 47) (35239, 25) (35941, 25) (6736, 26) (795315, 46)
(1211, 25) (1182, 25) (2607, 46) (35239, 24) (35941, 24) (6736, 25) (795315, 45)


### `labeled_data.csv` -> `supervised_label_cn7.csv`

In [14]:
print("labeled_data.shape", labeled.shape)
print("supervised_label_cn7.shape", supervised_cn7.shape)

labeled_data.shape (7996, 45)
supervised_label_cn7.shape (6736, 25)


In [15]:
print("Columns:", set(labeled.columns) & set(supervised_cn7.columns))
print("Only in labeled_data:", set(labeled.columns) - set(supervised_cn7.columns))
print("Only in supervised_label_cn7:", set(supervised_cn7.columns) - set(labeled.columns))

Columns: {'Clamp_Close_Time', 'Hopper_Temperature', 'PassOrFail', 'Cushion_Position', 'Average_Back_Pressure', 'Barrel_Temperature_5', 'Average_Screw_RPM', 'Mold_Temperature_4', 'Barrel_Temperature_3', 'Max_Back_Pressure', 'Max_Injection_Pressure', 'Barrel_Temperature_6', 'Injection_Time', 'Max_Injection_Speed', 'Barrel_Temperature_4', 'Max_Switch_Over_Pressure', 'Barrel_Temperature_2', 'Filling_Time', 'Plasticizing_Position', 'Mold_Temperature_3', 'Plasticizing_Time', 'Max_Screw_RPM', 'Cycle_Time', 'Clamp_Open_Position', 'Barrel_Temperature_1'}
Only in labeled_data: {'EQUIP_CD', 'Mold_Temperature_12', 'Barrel_Temperature_7', 'Mold_Temperature_9', 'PART_NAME', 'Mold_Temperature_6', 'Mold_Temperature_2', '_id', 'PART_FACT_SERIAL', 'TimeStamp', 'Switch_Over_Position', 'Reason', 'Mold_Temperature_11', 'Mold_Temperature_7', 'Mold_Temperature_5', 'Mold_Temperature_8', 'EQUIP_NAME', 'Mold_Temperature_10', 'Mold_Temperature_1', 'PART_FACT_PLAN_DATE'}
Only in supervised_label_cn7: set()


> `supervised_labeled_cn7.csv`에만 있는 컬럼은 없음

#### `labeled_data`에서 제거된 컬럼 확인

In [16]:
del_labeled = sorted(list(set(labeled.columns) - set(supervised_cn7.columns)))

for c in del_labeled:
    if c in ['_id', 'TimeStamp', 'PART_FACT_PLAN_DATE']:
        continue
    print(f"========== {c} ==========")
    print(labeled[c].value_counts())
    print("\n")

========== Barrel_Temperature_7 ==========
Barrel_Temperature_7
0.000000     7994
36.400002       2
Name: count, dtype: int64


========== EQUIP_CD ==========
EQUIP_CD
S14    7992
S01       2
S12       2
Name: count, dtype: int64


========== EQUIP_NAME ==========
EQUIP_NAME
650톤-우진2호기    7992
1800TON-우진       2
650톤-우진          2
Name: count, dtype: int64


========== Mold_Temperature_1 ==========
Mold_Temperature_1
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_10 ==========
Mold_Temperature_10
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_11 ==========
Mold_Temperature_11
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_12 ==========
Mold_Temperature_12
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_2 ==========
Mold_Temperature_2
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_5 ==========
Mold_Temperature_5
0.0    7996
Name: count, dtype: int64


========== Mold_Temperature_6 ===

- `_id` : 제조 공정 id **-> 불필요**
- `TimeStamp` : 시간 **-> 불필요**
- `PART_FACT_PLAN_DATE` : 생산을 지시한 날짜 **-> 불필요**
- `PART_FACT_SERIAL` : 생산을 지시한 품목에 대한 코드 부여 **-> 불필요**
- `PART_NAME` : 제품의 이름 **-> CN7과 RG3만 추출**
- `EQUIP_CD` : 생산한 사출기 호기 **-> S14만 선택**
- `EQUIP_NAME` : 생산한 사출기 모델명 **-> 650톤-우진2호기만 선택**
- `Reason` : 불량 유형 **-> 불필요**
- `Switch_Over_Position` : 고압(사출압)에서 보압으로 진행될 때의 위치 **-> 대부분 값이 0**
- `Barrel_Temperature_7` : 계량 및 사출 시 수지가 일정하게 용융(녹임)을 유지하기 위해 온도가 일정해야 한다 **-> 대부분 값이 0**
- `Mold_Temperature_1~2, 5~12` : 재료 주입구의 온도 **-> 모든 값이 0**

In [17]:
labeled.groupby('PART_NAME')[['EQUIP_CD', 'EQUIP_NAME']].value_counts()

PART_NAME                  EQUIP_CD  EQUIP_NAME
CN7 W/S SIDE MLD'G LH      S14       650톤-우진2호기    3365
CN7 W/S SIDE MLD'G RH      S14       650톤-우진2호기    3371
JX1 W/S SIDE MLD'G RH      S12       650톤-우진          2
RG3 MOLD'G W/SHLD, LH      S14       650톤-우진2호기     628
RG3 MOLD'G W/SHLD, RH      S14       650톤-우진2호기     628
SP2 CVR ROOF RACK CTR, RH  S01       1800TON-우진       2
Name: count, dtype: int64

#### `labeled_data.csv` 전처리 후 `supervised_label_cn7.csv`와 비교

In [18]:
# CN7 제품만 선택
# labeled_cn7 = labeled[labeled['PART_NAME'].isin(["CN7 W/S SIDE MLD'G LH", "CN7 W/S SIDE MLD'G RH"])].copy()
labeled_cn7 = labeled[labeled['PART_NAME'].str.contains("CN7")].copy()

# 불필요한 컬럼 제거
labeled_cn7.drop(columns=del_labeled, inplace=True)

# PassOrFail 인코딩
labeled_cn7['PassOrFail'] = labeled_cn7['PassOrFail'].map({'Y': 0, 'N': 1})
print("PassOrFail Unique Values:", labeled_cn7['PassOrFail'].unique(), end='\n\n')
print("Value Counts:\n", labeled_cn7['PassOrFail'].value_counts(), end='\n\n')

print("Equal:", supervised_cn7.equals(labeled_cn7))

PassOrFail Unique Values: [0 1]

Value Counts:
 PassOrFail
0    6697
1      39
Name: count, dtype: int64

Equal: False


In [19]:
print("Shape:", labeled_cn7.shape == supervised_cn7.shape)
print("Columns:", labeled_cn7.columns.equals(supervised_cn7.columns))
print("Index:", labeled_cn7.index.equals(supervised_cn7.index))
print("Values:", labeled_cn7.equals(supervised_cn7))

Shape: True
Columns: True
Index: False
Values: False


In [20]:
# labeled_data는 숫자형 컬럼의 소수점이 9자리까지 있고, supervised_label_cn7은 컬럼마다 다름

for decimals in range(10, 0, -1):

    df1 = labeled_cn7.copy()
    df2 = supervised_cn7.copy()

    num_cols = df1.select_dtypes(include="number").columns

    df1[num_cols] = df1[num_cols].round(decimals)
    df2[num_cols] = df2[num_cols].round(decimals)

    vc1 = df1.value_counts(dropna=False).sort_index()
    vc2 = df2.value_counts(dropna=False).sort_index()

    print(
        f"round({decimals}):",
        vc1.equals(vc2)
    )

round(10): False
round(9): False
round(8): False
round(7): False
round(6): True
round(5): True
round(4): True
round(3): True
round(2): True
round(1): True


In [21]:
# 소수점 6자리까지 반올림해서 다시 비교
df1 = labeled_cn7.copy()
df2 = supervised_cn7.copy()

num_cols = df1.select_dtypes(include="number").columns

df1[num_cols] = df1[num_cols].round(6)
df2[num_cols] = df2[num_cols].round(6)

vc1 = df1.value_counts(dropna=False).sort_index()
vc2 = df2.value_counts(dropna=False).sort_index()

print("Equal:", vc1.equals(vc2))

Equal: True


> `labeled_data.csv` -> `supervised_label_cn7.csv`가 맞음

#### supervised_label_rg3 ?

In [ ]:
# RG3 제품만 선택
supervised_rg3 = labeled[labeled['PART_NAME'].isin(["RG3 MOLD'G W/SHLD, LH", "RG3 MOLD'G W/SHLD, RH"])].copy()

# 불필요한 컬럼 제거
supervised_rg3.drop(columns=del_labeled, inplace=True)

# PassOrFail 인코딩
supervised_rg3['PassOrFail'] = supervised_rg3['PassOrFail'].map({'Y': 0, 'N': 1})
print("PassOrFail Unique Values:", supervised_rg3['PassOrFail'].unique(), end='\n')

# 확인
print("Shape:", supervised_rg3.shape, end='\n\n')
supervised_rg3.info()

PassOrFail Unique Values: [0 1]
Shape: (1256, 25)

<class 'pandas.DataFrame'>
Index: 1256 entries, 1211 to 5231
Data columns (total 25 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   PassOrFail                1256 non-null   int64  
 1   Injection_Time            1256 non-null   float64
 2   Filling_Time              1256 non-null   float64
 3   Plasticizing_Time         1256 non-null   float64
 4   Cycle_Time                1256 non-null   float64
 5   Clamp_Close_Time          1256 non-null   float64
 6   Cushion_Position          1256 non-null   float64
 7   Plasticizing_Position     1256 non-null   float64
 8   Clamp_Open_Position       1256 non-null   float64
 9   Max_Injection_Speed       1256 non-null   float64
 10  Max_Screw_RPM             1256 non-null   float64
 11  Average_Screw_RPM         1256 non-null   float64
 12  Max_Injection_Pressure    1256 non-null   float64
 13  Max_Switch_Over_Pressure 

In [ ]:
print(labeled_cn7.shape)
print("CN7 양품 수:", labeled_cn7[labeled_cn7['PassOrFail']==0].shape[0])
print("CN7 불량품 수:", labeled_cn7[labeled_cn7['PassOrFail']==1].shape[0])
print("="*20)
print(supervised_rg3.shape)
print("RG3 양품 수:", supervised_rg3[supervised_rg3['PassOrFail']==0].shape[0])
print("RG3 불량품 수:", supervised_rg3[supervised_rg3['PassOrFail']==1].shape[0])

(6736, 25)
CN7 양품 수: 6697
CN7 불량품 수: 39
(1256, 25)
RG3 양품 수: 1224
RG3 불량품 수: 32


### `moldset_labeled.csv` -> `moldset_labeled_cn7.csv`, `moldset_labeled_rg3.csv`


In [24]:
print("moldset_labeled.shape", moldset_labeled.shape)
print("moldset_labeled_cn7.shape", moldset_labeled_cn7.shape)
print("moldset_labeled_rg3.shape", moldset_labeled_rg3.shape)

moldset_labeled.shape (2607, 46)
moldset_labeled_cn7.shape (1211, 25)
moldset_labeled_rg3.shape (1182, 25)


In [25]:
set(moldset_labeled_cn7.columns) == set(moldset_labeled_rg3.columns)

True

In [26]:
print("Only in moldset_labeled:", set(moldset_labeled.columns) - set(labeled_cn7.columns))
print("Only in moldset_labeled_cn7/rg3:", set(moldset_labeled_cn7.columns) - set(moldset_labeled.columns))

Only in moldset_labeled: {'EQUIP_CD', 'Mold_Temperature_12', 'Barrel_Temperature_7', 'Mold_Temperature_9', 'PART_NAME', 'Mold_Temperature_6', 'Mold_Temperature_2', '_id', 'PART_NO', 'PART_FACT_SERIAL', 'TimeStamp', 'Switch_Over_Position', 'Reason', 'Mold_Temperature_11', 'Mold_Temperature_7', 'Mold_Temperature_5', 'Mold_Temperature_8', 'EQUIP_NAME', 'Mold_Temperature_10', 'Mold_Temperature_1', 'PART_FACT_PLAN_DATE'}
Only in moldset_labeled_cn7/rg3: set()


#### `moldset_labeled`에서 제거된 컬럼 확인

In [27]:
del_moldset = sorted(list(set(moldset_labeled.columns) - set(moldset_labeled_cn7.columns)))

for c in del_moldset:
    if c in ['_id', 'TimeStamp', 'PART_FACT_PLAN_DATE']:
        continue
    print(f"========== {c} ==========")
    print(moldset_labeled[c].value_counts())
    print("\n")

========== Barrel_Temperature_7 ==========
Barrel_Temperature_7
0.0    2607
Name: count, dtype: int64


========== EQUIP_CD ==========
EQUIP_CD
S14    2607
Name: count, dtype: int64


========== EQUIP_NAME ==========
EQUIP_NAME
650톤-우진2호기    2607
Name: count, dtype: int64


========== Mold_Temperature_1 ==========
Mold_Temperature_1
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_10 ==========
Mold_Temperature_10
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_11 ==========
Mold_Temperature_11
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_12 ==========
Mold_Temperature_12
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_2 ==========
Mold_Temperature_2
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_5 ==========
Mold_Temperature_5
0.0    2607
Name: count, dtype: int64


========== Mold_Temperature_6 ==========
Mold_Temperature_6
0.0    2607
Name: count, dtype: int64


========== Mold_Tem

- `TimeStamp` : 시간 **-> 불필요**
- `PART_FACT_PLAN_DATE` : 생산을 지시한 날짜 **-> 불필요**
- `PART_FACT_SERIAL` : 생산을 지시한 품목에 대한 코드 부여 **-> 불필요**
- `PART_NAME` : 제품의 이름 **-> CN7과 RG3으로 분리됨**
- `PART_NO` : 제품의 모델 코드 **-> CN7(86141AA000, 86131AA000)과 RG3(86131T1000, 86141T1000)으로 분리됨**
- `EQUIP_CD` : 생산한 사출기 호기 **-> S14만 존재**
- `EQUIP_NAME` : 생산한 사출기 모델명 **-> 650톤-우진2호기만 존재**
- `Reason` : 불량 유형 **-> 불필요, 결측치 많음**
- `Switch_Over_Position` : 고압(사출압)에서 보압으로 진행될 때의 위치 **-> 대부분 값이 0**
- `Barrel_Temperature_7` : 계량 및 사출 시 수지가 일정하게 용융(녹임)을 유지하기 위해 온도가 일정해야 한다 **-> 모든 값이 0**
- `Mold_Temperature_1~2, 5~12` : 재료 주입구의 온도 **-> 모든 값이 0**

In [28]:
moldset_labeled['PART_NO'].value_counts()

PART_NO
86141AA000    713
86131AA000    712
86131T1000    591
86141T1000    591
Name: count, dtype: int64

#### `moldset_labeled.csv` 전처리 후 `moldset_labeled_cn7/rg3.csv`와 비교

In [29]:
print("PassOrFail Unique Values:", moldset_labeled['PassOrFail'].unique())

PassOrFail Unique Values: [0 1]


In [30]:
# 제품 선택
moldset_processed_cn7 = moldset_labeled[moldset_labeled['PART_NO'].isin(["86131AA000", "86141AA000"])].copy()
moldset_processed_rg3 = moldset_labeled[moldset_labeled['PART_NO'].isin(["86131T1000", "86141T1000"])].copy()

# 불필요한 컬럼 제거
moldset_processed_cn7.drop(columns=del_moldset, inplace=True)
moldset_processed_rg3.drop(columns=del_moldset, inplace=True)

# PassOrFail 인코딩 불필요

print("moldset_processed_cn7.shape", moldset_processed_cn7.shape)
print("moldset_processed_rg3.shape", moldset_processed_rg3.shape)

moldset_processed_cn7.shape (1425, 25)
moldset_processed_rg3.shape (1182, 25)


> `moldset_labeled_cn7`은 (1211, 25)인데 `moldset_labeled_cn7.shape`는 (1425, 25)

In [31]:
print("CN7 양품 수:", moldset_processed_cn7[moldset_processed_cn7['PassOrFail']==0].shape[0])
print("CN7 불량품 수:", moldset_processed_cn7[moldset_processed_cn7['PassOrFail']==1].shape[0])
print("="*20)
print("RG3 양품 수:", moldset_processed_rg3[moldset_processed_rg3['PassOrFail']==0].shape[0])
print("RG3 불량품 수:", moldset_processed_rg3[moldset_processed_rg3['PassOrFail']==1].shape[0])

CN7 양품 수: 1398
CN7 불량품 수: 27
RG3 양품 수: 1157
RG3 불량품 수: 25


In [32]:
# moldset_labeled_rg3.csv만 비교해보기

# 가이드북대로 StandardScaler 적용
from sklearn.preprocessing import StandardScaler
for column in moldset_processed_rg3.columns:
    if column != 'PassOrFail':
        sc = StandardScaler()
        moldset_processed_rg3[column] = sc.fit_transform(moldset_processed_rg3[[column]])

# 값이 같아지는 소수점 자리수를 찾기 위해 15자리부터 1자리까지 반올림하여 비교
for decimals in range(15, 0, -1):

    df1 = moldset_labeled_rg3.copy()
    df2 = moldset_processed_rg3.copy()

    num_cols = df1.select_dtypes(include="number").columns

    df1[num_cols] = df1[num_cols].round(decimals)
    df2[num_cols] = df2[num_cols].round(decimals)

    vc1 = df1.value_counts(dropna=False).sort_index()
    vc2 = df2.value_counts(dropna=False).sort_index()

    print(
        f"round({decimals}):",
        vc1.equals(vc2)
    )

round(15): False
round(14): False
round(13): False
round(12): False
round(11): True
round(10): True
round(9): True
round(8): True
round(7): True
round(6): True
round(5): True
round(4): True
round(3): True
round(2): True
round(1): True


### `unlabeled_data.csv` 확인

In [33]:
unlabeled['PART_NAME'].value_counts()

PART_NAME
SP2 CVR ROOF RACK CTR, LH         118071
SP2 CVR ROOF RACK CTR, RH         114977
SP2 CAP-ROOF RACK(사출)              68342
LX2 W/S SIDE MLD'G LH              40647
CN7 W/S SIDE MLD'G RH              27885
SP2 PAD-ROOF RACK, RH              26248
CN7 W/S SIDE MLD'G LH              24662
LX2 W/S SIDE MLD'G RH              22354
RG3 MOLD'G W/SHLD, LH              20341
BC3 MOLD'G W/SHLD, RH              17808
RG3 MOLD'G W/SHLD, RH              17136
SP2 PAD-ROOF RACK, LH              15586
BC3 MOLD'G W/SHLD, LH              15368
RG3 MLD'G-RR GLS LWR CVR ASS'Y     14701
SK3 SIDE RAIL-ROOF RACK, RH        14676
RG3 MLD'G-RR GLS LWR CVR           14391
SK3 SIDE RAIL-ROOF RACK, LH        14372
CD SIDE MOLDING LH                 13063
SP2 SIDE RAIL-ROOF RACK, RH        12500
IK PNL C/PAD MAIN LHD              11718
PD W/S SIDE MLD'G LH               11562
CD SIDE MOLDING RH                 11069
JX1 W/S SIDE MLD'G LH              10796
LX2 W/S SIDE MLD'G RH(RHD)         10256
SP2 SI

In [34]:
print("Shape", unlabeled.shape)
print("Columns", unlabeled.columns)

Shape (795315, 45)
Columns Index(['_id', 'TimeStamp', 'PART_FACT_PLAN_DATE', 'PART_FACT_SERIAL',
       'PART_NO', 'PART_NAME', 'EQUIP_CD', 'EQUIP_NAME', 'ERR_FACT_QTY',
       'Injection_Time', 'Filling_Time', 'Plasticizing_Time', 'Cycle_Time',
       'Clamp_Close_Time', 'Cushion_Position', 'Switch_Over_Position',
       'Plasticizing_Position', 'Clamp_Open_Position', 'Max_Injection_Speed',
       'Max_Screw_RPM', 'Average_Screw_RPM', 'Max_Injection_Pressure',
       'Max_Switch_Over_Pressure', 'Max_Back_Pressure',
       'Average_Back_Pressure', 'Barrel_Temperature_1', 'Barrel_Temperature_2',
       'Barrel_Temperature_3', 'Barrel_Temperature_4', 'Barrel_Temperature_5',
       'Barrel_Temperature_6', 'Barrel_Temperature_7', 'Hopper_Temperature',
       'Mold_Temperature_1', 'Mold_Temperature_2', 'Mold_Temperature_3',
       'Mold_Temperature_4', 'Mold_Temperature_5', 'Mold_Temperature_6',
       'Mold_Temperature_7', 'Mold_Temperature_8', 'Mold_Temperature_9',
       'Mold_Temperatur

In [35]:
# 이후 준지도학습에서 moldset_labeled 데이터와 함께 사용해야 하므로 moldset_labeled와 컬럼 비교
print("Only in moldset_labeled:", set(moldset_labeled.columns) - set(unlabeled.columns))
print("Only in unlabeled:", set(unlabeled.columns) - set(moldset_labeled.columns))

Only in moldset_labeled: {'Reason', 'PassOrFail'}
Only in unlabeled: {'ERR_FACT_QTY'}


#### `ERR_FACT_QTY` 확인    
: 생산 시 발생한 불량의 개수

In [36]:
unlabeled['ERR_FACT_QTY'].value_counts()

ERR_FACT_QTY
0      185041
3       71100
2       59032
5       44944
4       38917
1       30754
12      29996
6       27150
7       23763
10      21499
9       20356
8       19347
15      14392
13      13875
11      13703
22      10035
14       9812
16       8260
19       8086
35       6738
17       6600
30       6600
24       6205
29       5800
34       5385
21       5150
62       4740
27       4620
48       4477
18       4124
63       3884
40       3808
23       3746
20       3363
81       3216
28       2958
90       2903
44       2836
54       2831
36       2674
58       2479
42       2374
130      2275
49       2125
31       2064
25       2016
55       1889
136      1833
37       1804
53       1699
177      1542
135      1536
80       1479
104      1463
76       1448
86       1381
72       1340
172      1260
45       1226
123      1200
43       1188
103      1186
77       1178
66       1165
154      1065
118      1002
98        971
57        906
32        837
153       804
52     

> `ERR_FACT_QTY`==0이면 `PassOrFail`==0(양품)과 같은거 아닌가?

#### 제품별로 나누기

In [43]:
unlabeled_cn7 = unlabeled[unlabeled['PART_NO'].isin(["86131AA000", "86141AA000"])].copy()
unlabeled_rg3 = unlabeled[unlabeled['PART_NO'].isin(["86131T1000", "86141T1000"])].copy()

unlabeled_pre_shape = [unlabeled_cn7.shape, unlabeled_rg3.shape]

print("unlabeled_processed_cn7.shape", unlabeled_cn7.shape)
print("unlabeled_processed_rg3.shape", unlabeled_rg3.shape)

unlabeled_processed_cn7.shape (52547, 45)
unlabeled_processed_rg3.shape (37477, 45)


In [37]:
print(unlabeled_cn7.info())
display(unlabeled_cn7.describe())

<class 'pandas.DataFrame'>
Index: 52547 entries, 114610 to 777220
Data columns (total 45 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   _id                       52547 non-null  str    
 1   TimeStamp                 52547 non-null  str    
 2   PART_FACT_PLAN_DATE       52547 non-null  str    
 3   PART_FACT_SERIAL          52547 non-null  int64  
 4   PART_NO                   52547 non-null  str    
 5   PART_NAME                 52547 non-null  str    
 6   EQUIP_CD                  52547 non-null  str    
 7   EQUIP_NAME                52547 non-null  str    
 8   ERR_FACT_QTY              52547 non-null  int64  
 9   Injection_Time            52547 non-null  float64
 10  Filling_Time              52547 non-null  float64
 11  Plasticizing_Time         52547 non-null  float64
 12  Cycle_Time                52547 non-null  float64
 13  Clamp_Close_Time          52547 non-null  float64
 14  Cushion_Position

,PART_FACT_SERIAL,ERR_FACT_QTY,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Switch_Over_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Barrel_Temperature_7,Hopper_Temperature,Mold_Temperature_1,Mold_Temperature_2,Mold_Temperature_3,Mold_Temperature_4,Mold_Temperature_5,Mold_Temperature_6,Mold_Temperature_7,Mold_Temperature_8,Mold_Temperature_9,Mold_Temperature_10,Mold_Temperature_11,Mold_Temperature_12
count,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.000000,52547.0,52547.0,52547.000000,52547.000000,52547.0,52547.0,52547.0,52547.0,52547.0,52547.0,52547.0,52547.0
mean,9.588863,28.881211,6.424337,4.218894,10.436911,50.905432,5.141773,359.175364,8.560623,52.581079,434.434858,56.389434,16.472642,156.852229,116.631891,99.514623,22.642434,27.234318,145.529239,267.087272,268.502999,265.401555,254.059596,233.254549,13.949301,57.740971,0.0,0.0,12.841428,14.730523,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
std,4.499093,38.978939,3.020516,1.444531,7.396193,13.369645,1.868678,317.320806,18.012402,13.087495,235.942618,29.437641,15.334763,145.803204,36.478173,29.089798,18.952520,29.014509,135.601133,11.345312,10.124143,8.909881,7.613118,6.257067,15.023507,10.945172,0.0,0.0,12.181921,14.162900,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
min,1.000000,0.000000,1.000000,0.300000,0.500000,0.000000,2.800000,15.800000,0.000000,32.180000,0.000000,25.100000,0.000000,0.000000,73.800000,68.800000,0.000000,0.000000,0.000000,244.200000,231.600000,239.100000,234.400000,219.200000,0.000000,34.300000,0.0,0.0,0.000000,0.000000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
25%,7.000000,5.000000,5.000000,3.890000,2.800000,36.600000,3.200000,17.300000,0.000000,40.300000,485.900000,36.500000,0.000000,0.000000,80.100000,69.900000,3.900000,0.000000,0.000000,264.600000,264.900000,264.800000,254.600000,229.800000,0.000000,57.800000,0.0,0.0,0.000000,0.000000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
50%,9.000000,12.000000,5.000000,4.490000,12.810000,59.520000,5.990000,653.410000,0.000000,52.660000,522.200000,49.900000,30.300000,290.300000,139.700000,106.700000,17.900000,12.200000,245.100000,264.900000,269.900000,269.400000,255.000000,234.700000,0.000000,61.600000,0.0,0.0,20.500000,22.000000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
75%,10.000000,34.000000,9.560000,5.000000,16.860000,63.600000,7.120000,653.560000,17.300000,68.470000,647.990000,56.100000,30.700000,292.400000,141.800000,136.300000,38.400000,59.300000,276.100000,275.400000,275.000000,270.500000,259.600000,239.900000,30.200000,65.200000,0.0,0.0,23.400000,26.700000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
max,27.000000,321.000000,11.400000,6.000000,97.080000,119.880000,7.270000,655.330000,655.330000,75.750000,652.000000,128.200000,62.900000,370.100000,187.000000,141.900000,97.400000,112.900000,287.500000,286.200000,285.800000,279.000000,267.200000,242.800000,31.600000,73.400000,0.0,0.0,38.600000,48.600000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0


In [38]:
print(unlabeled_rg3.info())
display(unlabeled_rg3.describe())

<class 'pandas.DataFrame'>
Index: 37477 entries, 99665 to 795309
Data columns (total 45 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   _id                       37477 non-null  str    
 1   TimeStamp                 37477 non-null  str    
 2   PART_FACT_PLAN_DATE       37477 non-null  str    
 3   PART_FACT_SERIAL          37477 non-null  int64  
 4   PART_NO                   37477 non-null  str    
 5   PART_NAME                 37477 non-null  str    
 6   EQUIP_CD                  37477 non-null  str    
 7   EQUIP_NAME                37477 non-null  str    
 8   ERR_FACT_QTY              37477 non-null  int64  
 9   Injection_Time            37477 non-null  float64
 10  Filling_Time              37477 non-null  float64
 11  Plasticizing_Time         37477 non-null  float64
 12  Cycle_Time                37477 non-null  float64
 13  Clamp_Close_Time          37477 non-null  float64
 14  Cushion_Position 

,PART_FACT_SERIAL,ERR_FACT_QTY,Injection_Time,Filling_Time,Plasticizing_Time,Cycle_Time,Clamp_Close_Time,Cushion_Position,Switch_Over_Position,Plasticizing_Position,Clamp_Open_Position,Max_Injection_Speed,Max_Screw_RPM,Average_Screw_RPM,Max_Injection_Pressure,Max_Switch_Over_Pressure,Max_Back_Pressure,Average_Back_Pressure,Barrel_Temperature_1,Barrel_Temperature_2,Barrel_Temperature_3,Barrel_Temperature_4,Barrel_Temperature_5,Barrel_Temperature_6,Barrel_Temperature_7,Hopper_Temperature,Mold_Temperature_1,Mold_Temperature_2,Mold_Temperature_3,Mold_Temperature_4,Mold_Temperature_5,Mold_Temperature_6,Mold_Temperature_7,Mold_Temperature_8,Mold_Temperature_9,Mold_Temperature_10,Mold_Temperature_11,Mold_Temperature_12
count,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.00000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.000000,37477.0,37477.0,37477.000000,37477.000000,37477.0,37477.0,37477.0,37477.0,37477.0,37477.0,37477.0,37477.0
mean,10.162953,44.417029,3.515335,2.778947,9.631108,53.469702,5.357147,405.104732,9.531962,48.223423,209.289599,86.657328,20.604886,179.079350,121.84281,97.877082,32.113184,36.580767,169.379563,273.179625,274.535830,269.174136,259.571657,235.694215,11.799690,58.029599,0.0,0.0,14.053569,15.813630,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
std,4.169985,67.320817,2.837536,2.087176,6.238300,12.758191,1.633415,310.466273,39.157049,7.846830,243.368693,41.854656,18.345535,145.771792,35.08609,52.450084,24.235511,33.234145,136.544147,12.796782,12.004843,9.526382,8.061772,4.830513,14.703037,9.782343,0.0,0.0,11.607758,13.580681,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
min,1.000000,0.000000,0.130000,0.000000,0.400000,0.000000,2.800000,0.000000,0.000000,33.350000,0.000000,0.000000,0.000000,0.000000,0.00000,69.100000,0.000000,0.000000,0.000000,244.200000,234.400000,239.100000,234.400000,219.500000,0.000000,33.200000,0.0,0.0,0.000000,0.000000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
25%,7.000000,3.000000,1.050000,0.930000,2.900000,38.900000,3.400000,18.700000,0.000000,40.500000,4.630000,39.900000,0.000000,0.000000,80.70000,70.000000,5.200000,0.000000,0.000000,264.800000,269.900000,269.800000,259.800000,234.800000,0.000000,58.000000,0.0,0.0,0.000000,0.000000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
50%,10.000000,13.000000,2.630000,1.650000,12.800000,61.760000,6.640000,654.250000,0.000000,52.540000,35.630000,79.500000,30.700000,290.400000,140.30000,111.700000,31.300000,58.200000,246.000000,265.400000,270.100000,270.200000,260.300000,235.100000,0.000000,61.300000,0.0,0.0,20.500000,22.000000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
75%,10.000000,55.000000,5.000000,5.000000,13.030000,63.600000,6.740000,654.280000,18.300000,53.570000,522.100000,127.500000,30.900000,290.600000,142.50000,116.000000,56.000000,60.800000,286.000000,285.100000,285.000000,275.200000,265.000000,239.900000,30.000000,64.100000,0.0,0.0,22.400000,24.000000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
max,21.000000,353.000000,11.390000,6.000000,38.710000,76.190000,15.100000,655.160000,655.320000,72.760000,654.990000,128.800000,62.900000,373.300000,186.90000,6553.100000,94.800000,136.000000,287.600000,286.700000,286.000000,278.100000,266.900000,241.000000,31.200000,71.200000,0.0,0.0,34.800000,45.900000,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0


#### moldset_labeled에서 drop한 컬럼 unlabeled에서도 확인

In [39]:
# unlabeled에는 없는 컬럼 제거
del_unlabeled = [c for c in del_moldset if c != 'Reason']
print(del_unlabeled)

['Barrel_Temperature_7', 'EQUIP_CD', 'EQUIP_NAME', 'Mold_Temperature_1', 'Mold_Temperature_10', 'Mold_Temperature_11', 'Mold_Temperature_12', 'Mold_Temperature_2', 'Mold_Temperature_5', 'Mold_Temperature_6', 'Mold_Temperature_7', 'Mold_Temperature_8', 'Mold_Temperature_9', 'PART_FACT_PLAN_DATE', 'PART_FACT_SERIAL', 'PART_NAME', 'PART_NO', 'Switch_Over_Position', 'TimeStamp', '_id']


In [40]:
# CN7
for c in del_unlabeled:
    if c in ['_id', 'TimeStamp', 'PART_FACT_PLAN_DATE']:
        continue
    print(f"========== {c} ==========")
    print(unlabeled_cn7[c].value_counts())
    print("\n")

========== Barrel_Temperature_7 ==========
Barrel_Temperature_7
0.0     28204
30.8     1581
30.9     1576
30.7     1545
30.3     1493
30.6     1414
30.4     1380
30.5     1373
31.0     1324
30.2     1126
30.1      989
30.0      906
29.9      841
29.0      688
29.7      661
29.1      657
29.4      648
29.5      646
29.2      639
29.8      637
29.3      620
29.6      614
28.9      559
28.8      472
31.1      431
28.7      377
28.6      305
31.3      157
28.5      142
31.2      135
31.4       99
31.5       81
28.4       50
31.6       26
26.8       12
28.2       11
28.3       10
27.4       10
27.2        9
27.5        9
28.0        9
27.6        8
26.5        8
28.1        7
27.0        6
27.1        6
27.7        6
27.3        5
27.8        5
27.9        5
26.9        4
26.3        4
26.4        4
26.1        3
26.2        3
26.7        2
25.7        2
26.6        2
25.8        1
Name: count, dtype: int64


========== EQUIP_CD ==========
EQUIP_CD
S14    35239
S06    17308
Name: count, dty

In [41]:
# RG3
for c in del_unlabeled:
    if c in ['_id', 'TimeStamp', 'PART_FACT_PLAN_DATE']:
        continue
    print(f"========== {c} ==========")
    print(unlabeled_rg3[c].value_counts())
    print("\n")

========== Barrel_Temperature_7 ==========
Barrel_Temperature_7
0.0     22787
30.9     1016
30.7     1001
30.8      990
30.6      944
30.5      907
30.4      860
31.0      802
30.3      779
30.2      751
30.1      671
30.0      578
29.9      482
29.8      455
28.9      404
29.7      384
29.6      377
29.1      375
28.8      368
29.4      348
29.0      347
29.5      343
29.3      338
29.2      313
28.7      238
31.1      211
28.6      172
28.5      145
28.4       46
28.3       21
31.2       12
28.2        3
26.7        2
26.8        2
26.9        2
27.0        2
27.1        1
Name: count, dtype: int64


========== EQUIP_CD ==========
EQUIP_CD
S14    35941
S06     1536
Name: count, dtype: int64


========== EQUIP_NAME ==========
EQUIP_NAME
650톤-우진2호기    35941
550TON-도시바     1536
Name: count, dtype: int64


========== Mold_Temperature_1 ==========
Mold_Temperature_1
0    37477
Name: count, dtype: int64


========== Mold_Temperature_10 ==========
Mold_Temperature_10
0    37477
Name: count,

> EQUIP : S14(650톤-우진2호기), S06(550TON-도시바) 두 가지 존재

In [44]:
# S14(650톤-우진2호기)만 선택
unlabeled_cn7 = unlabeled_cn7[unlabeled_cn7['EQUIP_CD']=='S14']
unlabeled_rg3 = unlabeled_rg3[unlabeled_rg3['EQUIP_CD']=='S14']

# 불필요한 컬럼 제거
unlabeled_cn7.drop(columns=del_unlabeled, inplace=True)
unlabeled_rg3.drop(columns=del_unlabeled, inplace=True)

print("unlabeled_cn7.shape ", unlabeled_pre_shape[0], "->", unlabeled_cn7.shape)
print("unlabeled_rg3.shape", unlabeled_pre_shape[1], "->", unlabeled_rg3.shape)

print("moldset_processed_cn7.shape", moldset_processed_cn7.shape)
print("moldset_processed_rg3.shape", moldset_processed_rg3.shape)

unlabeled_cn7.shape  (52547, 45) -> (35239, 25)
unlabeled_rg3.shape (37477, 45) -> (35941, 25)
moldset_processed_cn7.shape (1425, 25)
moldset_processed_rg3.shape (1182, 25)
